### PR TITLE
Add Travis configuration

### PR DESCRIPTION
**Pull request by kelunik**
`Registry` has already tests.

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -0,0 +1,24 @@
+language: php
+
+php:
+  - 5.5
+  - 5.6
+  - 7.0
+  - nightly
```
Comment (WyriHaximus) [inline .travis.yml:7]: What about HHVM? Do we want to test against it or and thus support it?

File: .travis.yml [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+   - php: nightly
+  fast_finish: true
+
+cache:
+  directories:
+    - $HOME/.composer/cache
+
+install:
+  - composer install
+  - composer show --installed
```
Comment (WyriHaximus) [inline .travis.yml]: IIRC the default behavior for `show` is now to only show installed packages removing the need to add `--installed`. (Assuming travis is up to date.)

File: .travis.yml (hunk 1 above)
Comment (kelunik) [inline .travis.yml:7]: Don't know. I personally don't care about HHVM.

File: .travis.yml (hunk 2 above)
Comment (kelunik) [inline .travis.yml]: Yes, and Travis also does `self-update` now automatically. Removed `self-update`, but not that.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,24 @@
...
+
+matrix:
+  allow_failures:
+   - php: nightly
+  fast_finish: true
+
+cache:
+  directories:
+    - $HOME/.composer/cache
+
```
Comment (prisis) [inline .travis.yml]: Add before_install: with - travis_retry composer self-update so that composer is up-to-date

File: .travis.yml (hunk 1 above)
Comment (PeeHaa) [inline .travis.yml:7]: How much effort would hhvm support need?

Comment (kelunik) [inline .travis.yml:7]: For this proposal probably none.

Comment (kelunik) [inline .travis.yml:7]: Should be fine now.

Comment (geggleto) [inline .travis.yml:7]: Not sure 5.5 is worth the efffort... its EOL is quite soon.It only has < 2 months of security patches.

Comment (bwoebi) [inline .travis.yml:7]: While that's true, we do not really require features which are only 5.6+. Thus, no harm in supporting 5.5; if we won't be able to anymore at some point, we can just drop it then.

Comment (kelunik) [inline .travis.yml:7]: @geggleto: We already discussed the minimum requirements. PHP 5.5 will be supported.

**Comment by AndrewCarterUK** [comment]
This looks good to me, @kelunik is it ready to merge?

**Comment by kelunik** [comment]
Yes.

**Pull request merged**